tap_appsflyer/client.py

```diff
@@ -0,0 +1,139 @@
+from typing import Any, Dict, Mapping, Optional, Tuple
+
+import backoff
+import requests
+from requests import session
+from requests.exceptions import Timeout, ConnectionError, ChunkedEncodingError
+from singer import get_logger, metrics, utils
+
+from tap_appsflyer.exceptions import (
+    ERROR_CODE_EXCEPTION_MAPPING,
+    appsflyerError,
+    appsflyerBackoffError,
+)
+
+LOGGER = get_logger()
+REQUEST_TIMEOUT = 300
+
+
+def raise_for_error(response: requests.Response) -> None:
+    """Raises the associated response exception. Takes in a response object,
+    checks the status code, and throws the associated exception based on the
+    status code.
+
+    :param resp: requests.Response object
+    """
+    try:
+        response_json = response.json()
+    except Exception:
+        response_json = {}
+    if response.status_code != 200:
+        if response_json.get("error"):
+            message = "HTTP-error-code: {}, Error: {}".format(
+                response.status_code, response_json.get("error")
+            )
+        else:
+            message = "HTTP-error-code: {}, Error: {}".format(
+                response.status_code,
+                response_json.get(
+                    "message",
+                    ERROR_CODE_EXCEPTION_MAPPING.get(response.status_code, {}).get(
+                        "message", "Unknown Error"
+                    ),
+                ),
+            )
```

Comment:
Use f-string to offer better redeability.

```suggestion
        if response_json.get("error"):
            message = f"HTTP-error-code: {response.status_code}, Error: {response_json.get("error")}"
        else:
            error_message = ERROR_CODE_EXCEPTION_MAPPING.get(response.status_code, {}).get(
    "message", "Unknown Error"
)
            message = f"HTTP-error-code: {response.status_code}, Error: {response_json.get('message', error_message)}"
```